zenoh-flow/src/model/mod.rs

```diff
@@ -12,6 +12,43 @@
 //   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
 //
 
+use std::{path::PathBuf, str::FromStr};
+
+use crate::{bail, prelude::ErrorKind, zfresult::ZFError};
+
+use self::registry::NodeKind;
+
 pub mod descriptor;
 pub mod record;
 pub mod registry;
+
+/// The middleware used for the builtin sources and sink.
+#[derive(Debug)]
+pub(crate) enum Middleware {
+    Zenoh,
+}
+
+impl FromStr for Middleware {
+    type Err = ZFError;
+
+    fn from_str(s: &str) -> std::result::Result<Self, Self::Err> {
+        let s = s.to_lowercase();
+        if s == "zenoh" {
+            return Ok(Self::Zenoh);
+        }
+        bail!(ErrorKind::ParsingError, "{s} is not a valid middleware!")
```

Comment:
```suggestion
        bail!(ErrorKind::ParsingError, "Unsupported middleware: '{s}'. Currently supported middlewares: 'zenoh'.")
```

Or a similar message that lists the supported middleware.